docs/guides/3_validating_public_input.md

```diff
@@ -0,0 +1,201 @@
+            (uint32(uint8(data[1])) << 8) |
+            (uint32(uint8(data[2])) << 16) |
+            (uint32(uint8(data[3])) << 24);
+
+        uint32 second = uint32(uint8(data[4])) |
+            (uint32(uint8(data[5])) << 8) |
+            (uint32(uint8(data[6])) << 16) |
+            (uint32(uint8(data[7])) << 24);
+
+        return (first, second);
+    }
+}
+```
+
+### Explanation
+
+1. **Verification Key Check:** The contract first checks if the verification key commitment matches the Fibonacci Image ID commitment.
+
+```solidity
+require(
+            fibonacciImageIdCommitment == provingSystemAuxDataCommitment,
+            "Image ID doesn't match"
+        );
+```
+
+2. **Public Input Commitment Validation:** The contract validates that the public input commitment matches the keccak256 hash of the journalBytes. It then extracts the last two Fibonacci numbers from the journalBytes and emits an event.
+
+```solidity
+require(
+    pubInputCommitment == keccak256(abi.encodePacked(journalBytes)),
+    "Fibonacci numbers don't match with public input"
+);
+
+(uint32 fibN, uint32 fibNPlusOne) = bytesToTwoUint32(journalBytes);
+
+emit FibonacciNumbers(fibN, fibNPlusOne);
+```
+
+3. **Static Call to AlignedServiceManager**: The contract makes a static call to the `AlignedServiceManager` contract to check if the proof was verified in Aligned.
+
+```solidity
+(
+    bool callWasSuccessful,
+    bytes memory proofIsIncluded
+) = alignedServiceManager.staticcall(
+    abi.encodeWithSignature(
+        "verifyBatchInclusion(bytes32,bytes32,bytes32,bytes20,bytes32,bytes,uint256)",
+        proofCommitment,
+        pubInputCommitment,
+        provingSystemAuxDataCommitment,
+        proofGeneratorAddr,
+        batchMerkleRoot,
+        merkleProof,
+        verificationDataBatchIndex
+    )
+);
+
+require(callWasSuccessful, "static_call failed");
+```
+
+4. **Bytes to two `uint32` conversion:** A helper function to convert a byte array into two `uint32` numbers, used for extracting the last two Fibonacci numbers from the `journalBytes`.
+
+```solidity
+function bytesToTwoUint32(
+    bytes memory data
+) public pure returns (uint32, uint32) {
+    require(data.length >= 8, "Input bytes must be at least 8 bytes long");
+
+    uint32 first = uint32(uint8(data[0])) |
+        (uint32(uint8(data[1])) << 8) |
+        (uint32(uint8(data[2])) << 16) |
+        (uint32(uint8(data[3])) << 24);
+
+    uint32 second = uint32(uint8(data[4])) |
+        (uint32(uint8(data[5])) << 8) |
+        (uint32(uint8(data[6])) << 16) |
+        (uint32(uint8(data[7])) << 24);
+
+    return (first, second);
+}
+```
+
+To deploy the contract, first you will need to set up the `.env` file in the contracts folder with the following variables:
+
+```
+RPC_URL= #You can use publicnode RPC: https://ethereum-holesky-rpc.publicnode.com
+PRIVATE_KEY=
+ALIGNED_SERVICE_MANAGER_ADDRESS= #0x58F280BeBE9B34c9939C3C39e0890C81f163B623 for Holesky
```

Comment:
```suggestion
RPC_URL=<rpc_url> #You can use publicnode RPC: https://ethereum-holesky-rpc.publicnode.com
PRIVATE_KEY=<private_key>
ALIGNED_SERVICE_MANAGER_ADDRESS=<service_manager_address> #0x58F280BeBE9B34c9939C3C39e0890C81f163B623 for Holesky
```

Reply:
Solved in [bc4fa19](https://github.com/yetanotherco/aligned_layer/pull/648/commits/bc4fa195fe395fd6b590437e330ee8943ec792be)